locust/test/test_main.py

```diff
@@ -144,20 +250,28 @@ def my_task(self):
             try:
                 with open("locust.conf", "w") as conf_file:
                     conf_file.write("custom-string-arg config_file_value")
+
+                port = get_free_tcp_port()
+
                 proc = subprocess.Popen(
-                    ["locust", "-f", file_path, "--autostart"],
-                    stdout=PIPE,
-                    stderr=PIPE,
+                    ["locust", "-f", file_path, "--autostart", "--web-port", str(port), "--config", "locust.conf"],
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
                     text=True,
                 )
-                gevent.sleep(1)
+
+                poll_until(lambda: web_interface_ready("localhost", port), timeout=20)
+
             finally:
-                os.remove("locust.conf")
+                if os.path.exists("locust.conf"):
```

Comment:
this isn't necessarily an improvement. if locust.conf doesnt exist, something is really broken and we'd wanna crash ASAP :)